src/main/java/study/car/race/RacingCars.java

```diff
@@ -0,0 +1,23 @@
+package study.car.race;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RacingCars {
+
+    private final List<Car> cars = new ArrayList<>();
+
+    public RacingCars(int carCount) {
+        for (int i = 0; i < carCount; i++) {
+            cars.add(new Car());
+        }
+    }
+
+    public void race() {
+        cars.forEach(car -> car.moveOrStop(RandomUtils.nextInt()));
+    }
+
+    public List<Car> result() {
+        return cars;
```

Comment:
이렇게 그대로 반환하면 위 변조에 취약할 수 있습니다. 
```suggestion
        return Collections.unmodifiableList(cars);
```

혹은 
```suggestion
        return new ArrayList<>(challengers);;
```

이런식으로 하면 수정이 불가능하게 되거나 아예 새로운 리스트를 만들기에 
반환된 자동차 목록을 추가,삭제하더라도 일급 콜렉션에는 영향을 안끼치겠죠 ㅎ